### PR TITLE
ZCS-11426 Need new column in zimbra.volume table

### PR DESCRIPTION
**Pull request by inpati**
Problem Statement
Need new column in zimbra.volume table

Solution
Added new migration file which shall allow us to create a new column in zimbra.volume table

### CODE REVIEW

File: src/db/migration/migrate20220525-Volume.pl [hunk 1]
```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/perl
+# 
+# ***** BEGIN LICENSE BLOCK *****
+# Zimbra Collaboration Suite Server
+# Copyright (C) 2005, 2007, 2008, 2009, 2010, 2013, 2014, 2016 Synacor, Inc.
```
Comment (dasiyogesh) [inline src/db/migration/migrate20220525-Volume.pl:5]: replace all the years with just  2022, when adding new file add just the current year and while modifying existing file add the current year if not in the list

File: src/db/migration/migrate20220525-Volume.pl [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    my $sth = $dbh->prepare ($sql);
+    $sth->execute();
+
+    # print 'path' column from table volume
+    while(my @row = $sth->fetchrow_array()) {
+       printf("%s\t\n",$row[0]);
+       my $varBool = $PREFIX eq substr($row[0],0,length($PREFIX));
+       # printf(" --> do drive starts with prefix ? [%s]\n",$varBool);
+
+        if($varBool eq 1)
```
Comment (dasiyogesh) [inline src/db/migration/migrate20220525-Volume.pl:84]: add space after if

File: src/db/migration/migrate20220525-Volume.pl [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        sleep 1;
+    }
+
+    # prepare and execute query
+    my $sql = "SELECT path FROM volume";
+    my $sth = $dbh->prepare ($sql);
+    $sth->execute();
+
+    # print 'path' column from table volume
+    while(my @row = $sth->fetchrow_array()) {
```
Comment (dasiyogesh) [inline src/db/migration/migrate20220525-Volume.pl:79]: add space after while

File: src/db/migration/migrate20220525-Volume.pl [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+# Verify Schema Version Number
+Migrate::verifySchemaVersion(111);
+
+addStoreTypeColumn();
+updateStoreTypeColumn();
+
+# Update Schema Version Number
+Migrate::updateSchemaVersion(111, 112);
```
Comment (dasiyogesh) [inline src/db/migration/migrate20220525-Volume.pl:48]: looks like DB version is incorrect, in earlier commits version is 114

File: src/db/migration/migrate20220525-Volume.pl [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+exit(0);
+
+# Function to add 'store_type' column
+sub addStoreTypeColumn() {
+    my $sql = <<VOLUME_ADD_COLUMN_EOF;
+ALTER TABLE volume ADD COLUMN IF NOT EXISTS store_type TINYINT NOT NULL DEFAULT '1' COMMENT '1 for onstore and 2 for s3 bucket';
```
Comment (deepakzimb) [inline src/db/migration/migrate20220525-Volume.pl:55]: Default is 1, so make sure in ticket description it should be 1

**Review by deepakzimb — COMMENTED** [review]
Please confirm about zm-build repo changes also

File: src/db/migration/migrate20220525-Volume.pl (hunk 4 above)
Comment (inpati) [reply src/db/migration/migrate20220525-Volume.pl:48]: Ok. DB version updated to 115

File: src/db/migration/migrate20220525-Volume.pl (hunk 2 above)
Comment (inpati) [reply src/db/migration/migrate20220525-Volume.pl:84]: Done

File: src/db/migration/migrate20220525-Volume.pl (hunk 1 above)
Comment (inpati) [reply src/db/migration/migrate20220525-Volume.pl:5]: Done

File: src/db/migration/migrate20220525-Volume.pl (hunk 5 above)
Comment (inpati) [reply src/db/migration/migrate20220525-Volume.pl:55]: Done

**Comment by dasiyogesh** [comment]
@inpati review comments are showing resolved but the change commits are not showing.

**Comment by inpati** [comment]
> @inpati review comments are showing resolved but the change commits are not showing.

https://github.com/Zimbra/zm-db-conf/pull/25/commits/f59a78c4aac40e4459d1c4cca4583c6fa5f5edde

File: src/db/migration/migrate20220525-Volume.pl [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+       # printf(" --> do drive starts with prefix ? [%s]\n",$varBool);
+
+        if ($varBool eq 1) {
+            # Migrate::log("varBool equals 1");
+            my $sql = "UPDATE volume SET store_type=2 WHERE path='$row[0]'";
+            my $sth = $dbh->prepare ($sql);
+            $sth->execute();
+            $sth->finish();
+        }
+        else {
```
Comment (dasiyogesh) [inline src/db/migration/migrate20220525-Volume.pl:91]: move else to above line as
if () {
} else {
}

File: src/db/migration/migrate20220525-Volume.pl [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    my $sth = $dbh->prepare ($sql);
+    $sth->execute();
+
+    # print 'path' column from table volume
+    while (my @row = $sth->fetchrow_array()) {
+       printf("%s\t\n",$row[0]);
+       my $varBool = $PREFIX eq substr($row[0],0,length($PREFIX));
+       # printf(" --> do drive starts with prefix ? [%s]\n",$varBool);
+
+        if ($varBool eq 1) {
```
Comment (dasiyogesh) [inline src/db/migration/migrate20220525-Volume.pl:84]: indentation seems to be bit off from here below

**Review by dasiyogesh — APPROVED** [review]

**Review by zimsuchitgupta — APPROVED** [review]

**Review by shrutig0510 — APPROVED** [review]

File: src/db/migration/migrate20220525-Volume.pl [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    # prepare and execute query
+    my $sql = "SELECT path FROM volume";
+    my $sth = $dbh->prepare ($sql);
+    $sth->execute();
+
+    # print 'path' column from table volume
+    while (my @row = $sth->fetchrow_array()) {
+        printf("%s\t\n",$row[0]);
+        my $varBool = $PREFIX eq substr($row[0],0,length($PREFIX));
+        # printf(" --> do drive starts with prefix ? [%s]\n",$varBool);
```
[the remark this answers is not in the document]
Comment (inpati) [reply src/db/migration/migrate20220525-Volume.pl]: As discussed removed dead code and else part

**Pull request merged**